CMakeLists.txt

```
@@ -0,0 +1,188 @@
+  src/StretchCalculator.cpp
+  src/StretcherChannelData.cpp
+  src/StretcherImpl.cpp
+  src/StretcherProcess.cpp
+  src/audiocurves/CompoundAudioCurve.cpp
+  src/audiocurves/ConstantAudioCurve.cpp
+  src/audiocurves/HighFrequencyAudioCurve.cpp
+  src/audiocurves/PercussiveAudioCurve.cpp
+  src/audiocurves/SilentAudioCurve.cpp
+  src/audiocurves/SpectralDifferenceAudioCurve.cpp
+  src/base/Profiler.cpp
+  src/dsp/AudioCurveCalculator.cpp
+  src/dsp/FFT.cpp
+  src/dsp/Resampler.cpp
+  src/rubberband-c.cpp
+  src/system/Allocators.cpp
+  src/system/Thread.cpp
+  src/system/VectorOpsComplex.cpp
+  src/system/sysutils.cpp
+)
+
+#
+# Dependencies
+#
+
+# Threads
+find_package(Threads REQUIRED)
+target_link_libraries(rubberband PRIVATE Threads::Threads)
+
+# FFTW / KissFFT
+if(USE_FFTW)
+  message(STATUS "Using FFT library: fftw")
+  find_path(FFTW3_INCLUDEDIR fftw3.h REQUIRED)
+  find_library(FFTW3_LIBRARY fftw3 REQUIRED)
+  target_include_directories(rubberband PRIVATE ${FFTW3_INCLUDEDIR})
+  target_link_libraries(rubberband PUBLIC ${FFTW3_LIBRARY})
+  target_compile_definitions(rubberband PRIVATE HAVE_FFTW3 FFTW_DOUBLE_ONLY)
+else()
+  message(STATUS "Using resampling library: kissfft")
+  target_compile_definitions(rubberband PRIVATE USE_KISSFFT)
+  target_sources(rubberband PRIVATE src/kissfft/kiss_fft.c src/kissfft/kiss_fftr.c)
+endif()
+
+# libsamplerate / Speex
+if(USE_LIBSAMPLERATE)
+  message(STATUS "Using resampling library: libsamplerate")
+  find_path(SAMPLERATE_INCLUDEDIR samplerate.h REQUIRED)
+  find_library(SAMPLERATE_LIBRARY samplerate REQUIRED)
+  target_include_directories(rubberband PRIVATE ${SAMPLERATE_INCLUDEDIR})
+  target_link_libraries(rubberband PUBLIC ${SAMPLERATE_LIBRARY})
+  target_compile_definitions(rubberband PRIVATE HAVE_LIBSAMPLERATE)
+else()
+  message(STATUS "Using resampling library: speex")
+  target_compile_definitions(rubberband PRIVATE USE_SPEEX)
+  target_sources(rubberband PRIVATE src/speex/resample.c)
+endif()
+
+# Accelerate framework
+if(APPLE)
+  find_library(ACCELERATE_FRAMEWORK NAMES Accelerate REQUIRED)
+  target_link_libraries(rubberband PUBLIC ${ACCELERATE_FRAMEWORK})
+endif()
+
+#
+# Handle general flags, include paths, etc.
+#
+include(CheckCXXCompilerFlag)
+
+if(OPTIMIZE)
+  check_cxx_compiler_flag("-march=native" MARCH_NATIVE_SUPPORTED)
+  if(MARCH_NATIVE_SUPPORTED AND NOT MSVC)
+    target_compile_options(rubberband PRIVATE -march=native)
+  endif()
+endif()
+
+target_include_directories(rubberband
+  PUBLIC
+      $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/rubberband>
+      $<INSTALL_INTERFACE:rubberband>
+  PRIVATE
+      $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/>
+      $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/src>
+)
+
+target_compile_definitions(rubberband
+  PRIVATE
+    $<$<BOOL:${CMAKE_USE_PTHREADS_INIT}>:USE_PTHREADS>
+    MALLOC_IS_ALIGNED
+    NO_THREAD_CHECKS
+    NO_TIMING
+)
+
+if(APPLE)
+  target_compile_options(rubberband PRIVATE -mmacosx-version-min=10.11)
```

```suggestion
  if(CMAKE_OSX_DEPLOYMENT_TARGET VERSION_LESS 10.11)
    message(FATAL_ERROR "macOS deployment target must be >= 10.11)
  endif()
```
I don't think the compiler flag should be set explicitly. I'm not sure, but I think newer deployment targets may allow more optimizations. The user should be able to set newer deployment targets above the minimum if they want.

---

https://cmake.org/cmake/help/latest/variable/CMAKE_OSX_DEPLOYMENT_TARGET.html#variable:CMAKE_OSX_DEPLOYMENT_TARGET
> It is intended to be set locally by the user creating a build tree.